plugin/plugins/filesystem/nodes/root.go

```diff
@@ -0,0 +1,139 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/djdv/p9/p9"
+	cid "github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/multiformats/go-multihash"
+)
+
+var _ p9.File = (*RootIndex)(nil)
+
+func newRootPath(path string) corepath.Resolved {
+	return rootNode(path)
+}
+
+type rootNode string
+
+func (rn rootNode) String() string { return string(rn) }
+func (rootNode) Namespace() string { return nRoot }
+func (rootNode) Mutable() bool     { return true }
+func (rootNode) IsValid() error    { return nil } //TODO: should return ENotImpl
```

Comment:
IMO, this is a valid path.

Reply:
I'd agree with that. It's a valid path according to our system, even if it's not for the CoreAPI.
So long as we keep it local, it should be fine.